### PR TITLE
Form proper labels for Category/Project update link

### LINKED ISSUE #1421: Incorrect project type strings returned for sidebar and project dropdown
### Describe the Bug

Something we missed in #1380 it seems: the correct project type is no longer used in the sidebar links. When viewing the links in a category, the update view link is displayed as `Update Project`.

### How to Reproduce

Create a category, navigate into the category and observe the `Update Project` link in the sidebar instead of `Update Category`.

### Expected Behavior

In case of a category, the link should display `Update Category`. For a project, it should say `Update Project`.

Furthermore, we should be using `get_display_name()` to return the project and category labels as configured by the site using SODAR Core. In some sites based on this platform, different terms are used for projects and categories.

All this logic was correctly present in the original implementation in the templates, but for some reason hardcoded strings are currently used in `AppLinkContent.get_project_app_links()`. The original behaviour should be replicated.

Tests will of course need to be updated to catch and assert this correct behaviour.

### Screenshots

2024-04_sidebar_project_type_bug

### PR DESCRIPTION
**Pull request by gromdimon**
Issue: #1421

### CODE REVIEW

File: projectroles/utils.py [hunk 1]
```diff
@@ -191,7 +191,7 @@ def get_project_app_links(
                         'projectroles:detail',
                         kwargs={'project': project.sodar_uuid},
                     ),
-                    'label': 'Project Overview',
+                    'label': 'Overview',
```
Comment (mikkonie) [inline projectroles/utils.py]: This one is just "Overview" now, the project/category title is missing.

File: projectroles/utils.py [hunk 2]
```diff
@@ -249,7 +249,7 @@ def get_project_app_links(
                             'projectroles:update',
                             kwargs={'project': project.sodar_uuid},
                         ),
-                        'label': 'Update Project',
+                        'label': f'Update {get_display_name(project.type)}',
```
Comment (mikkonie) [inline projectroles/utils.py]: You need to supply `title=True` as in the original implementation to ensure the name is returned capitalized. This was present in the original implementation. Please test these features to ensure they appear correct before submitting a pr.

File: projectroles/utils.py [hunk 3]
```diff
@@ -249,7 +249,7 @@ def get_project_app_links(
...
                         'icon': 'mdi:lead-pencil',
                         'active': self._is_active_projectroles(
                             link_names=['update'],
```
Comment (mikkonie) [inline projectroles/utils.py:256]: For all the links after this point in the code, all the labels are still provided as hardcoded text. Please go through all the changes required carefully before submitting.

File: projectroles/tests/test_templatetags.py [hunk 4]
```diff
@@ -690,7 +691,7 @@ def test_get_sidebar_links_project_detail_view(self):
                 {
                     'name': 'project-update',
                     'url': f'/project/project/update/{self.project.sodar_uuid}',
-                    'label': 'Update Project',
+                    'label': f'Update {get_display_name(self.project.type)}',
```
Comment (mikkonie) [inline projectroles/tests/test_templatetags.py]: Probably better to check for a literal string here?

**Review by mikkonie — CHANGES_REQUESTED** [review]

File: projectroles/utils.py (hunk 2 above)
Comment (gromdimon) [reply projectroles/utils.py]: I'm sorry, that I overlooked it! Now it should be fixed

**Comment by gromdimon** [comment]
It looks like some 'label' lines are now too long. Although Black and Flake8 do not raise any issues, this definitely doesn't follow conventions. There are multiple ways to solve this problem: reformatting the string using parentheses (), using the '+' syntax, adding a helper method, or introducing some temporary variables before appending the link.

The question now is: how should I address this issue?

**Comment by mikkonie** [comment]
> It looks like some 'label' lines are now too long. Although Black and Flake8 do not raise any issues, this definitely doesn't follow conventions. There are multiple ways to solve this problem: reformatting the string using parentheses (), using the '+' syntax, adding a helper method, or introducing some temporary variables before appending the link.
> 
> The question now is: how should I address this issue?

This is a good point. You have "wrapped" the code inside the format string and strings are exempt from line length formatting in black. It is possible to do the following:

```
'label': f'Create '
'{get_display_name(PROJECT_TYPE_PROJECT, title=True)}',
```

That reduces line length and is accepted by black. Probably the best way unless you want to use the old style `format()` or get the title into a variable before inserting it into the string.

File: projectroles/utils.py [hunk 5]
```diff
@@ -191,7 +191,7 @@ def get_project_app_links(
                         'projectroles:detail',
                         kwargs={'project': project.sodar_uuid},
                     ),
-                    'label': 'Project Overview',
+                    'label': f'{get_display_name(project.type, title=True)} Overview',
```
Comment (mikkonie) [inline projectroles/utils.py]: We could get the string for the current project's type into a variable and use it both here and in the "Update Project" link. That would save us from calling the same method twice with identical input.

**Review by mikkonie — CHANGES_REQUESTED** [review]
I had one minor comment, plus I gave a hint on how to fix the line length issue in the pull request comments. Otherwise this looks fine and once these small things are fixed, I'll merge.

**Review by mikkonie — APPROVED** [review]

**Pull request merged**